client/allocrunner/alloc_runner_hooks.go

```diff
@@ -77,7 +77,8 @@ func (ar *allocRunner) initRunnerHooks() {
 	// directory path exists for other hooks.
 	ar.runnerHooks = []interfaces.RunnerHook{
 		newAllocDirHook(hookLogger, ar.allocDir),
-		newDiskMigrationHook(hookLogger, ar.prevAllocWatcher, ar.allocDir),
+		newPreemptionHook(hookLogger, ar.prevAllocWatcher),
```

Comment:
The alloc runner already invokes your [wait](https://github.com/hashicorp/nomad/pull/4963/files#diff-ebb76b5e0ecb0955643d61deb3b99a41R32) method so I don't follow why we need to create a hook that also invokes the same Wait method. (unless I misunderstood the control flow here)

Reply:
It only invokes wait in the diskMigrationHook for a PreviousAllocation that might need migration